lib/native/aws_iot.ts

```diff
@@ -178,7 +179,11 @@ export class AwsIotMqttConnectionConfigBuilder {
                     await aws_sign_request(request, signing_config as AwsSigningConfig);
                     done();
                 } catch (error) {
-                    done(error);
+                    if (error instanceof CrtError) {
+                        done(error.error_code);
+                    } else {
+                        done(3); /* TODO: AWS_ERROR_UNKNOWN */
```

Comment:
Question: Should we remove this TODO?

Reply:
Well, I don't want to make modeling native error codes part of this PR but I also don't want to completely "forget" that this is a temporary work-around.

Reply:
Ah okay. I think for now, it's fine to leave it to get these fixes out.
Maybe I'll look at adding this as part of my on-call. 👍